packages/tldraw/src/lib/tools/SelectTool/childStates/Idle.ts

```diff
@@ -505,13 +505,8 @@ export class Idle extends StateNode {
 
 		// todo: Extract into general hit test for arrows
 		if (this.editor.isShapeOfType<TLArrowShape>(shape, 'arrow')) {
-			// How should we handle multiple labels? Do shapes ever have multiple labels?
-			const labelGeometry = this.editor.getShapeGeometry<Group2d>(shape).children[1]
-			// Knowing what we know about arrows... if the shape has no text in its label,
-			// then the label geometry should not be there.
-			if (labelGeometry && pointInPolygon(pointInShapeSpace, labelGeometry.vertices)) {
-				return true
-			}
+			const arrowShapeUtil = this.editor.getShapeUtil(shape) as ArrowShapeUtil
+			return arrowShapeUtil.isOverArrowLabelTest(shape, pointInShapeSpace)
```

Comment:
Given the choice, I'd prefer the original implementation. It's a little fragile but acceptable, and better IMO than extending the public interface of a shapeutil.

Reply:
i agree with David that we should keep the arrow logic together with other arrow logic - otherwise it might start getting out of sync.